src/UriId.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Assertis\Util;
+
+use InvalidArgumentException;
+
+/**
+ * @author Michał Tatarynowicz <michal.tatarynowicz@assertis.co.uk>
+ */
+abstract class UriId extends Id
+{
+    const REGEX = '|^(/[a-z0-9\-_\.\?\=]+)+$|';
+
+    /**
+     * @return string
+     */
+    protected function getValueRegex()
+    {
+        return self::REGEX;
+    }
+
+    /**
+     * @return string
+     */
+    protected function assertIsValidUri($value)
+    {
+        if (preg_match($this->getValueRegex(), $value)) {
```

Comment:
So that it can be overridden in child classes. I don't like overriding class constants as a way of creating an interface.